equinox/_module/README.md

```diff
@@ -0,0 +1,35 @@
+# `equinox.Module`
+
+This is the core of the library: subclassing `equinox.Module` ensures that your class is both a PyTree and a frozen dataclass.
+
+The above would be just a couple of lines of code, of course (and this idea exists in numerous libraries, from both before and after Equinox's inception). What else is going on here? Or in other words, why do we think Equinox's implementation is probably the most complete amongst this class of ideas?
+
+## Safety / no footguns:
+
+- Equinox has a lot of nice error messages for every kind of edge case we can think of:
+  - Accidentally marking arrays as static fields;
+  - Accidentally creating cycles via method assignment;
+  - Accidentally closing over arrays;
+  - Accidentally ignoring `__post_init__` because an `__init__` method already exists.
+- Good behaviour around edge cases:
+  - Bound methods like `foo = SomeModule().some_method` are also pytrees, so that things like `jax.jit(...)(foo)` work correctly.
+  - Subclassing propagates `__init__` methods, `__doc__`strings.
+  - Mutating attributes during `__init__` is totally fine because we don't do any `__setattr__` magic.
```

Comment:
Wouldn't be better to include, for each line (at least non-trivial ones), a small code example?

I think that not all users may see how we could create cycles via method assignments, or *closing over arrays* (especially for non-native English speakers, like me, I had to check on SO to know what it meant ^^').

Reply:
I considered this and left it off, in the interests of a smaller document.

This document isn't meant to be public anywhere, it's just meant as an easter egg summary for anyone curious enough to go poking through the codebase!